firestore.rules

```diff
@@ -1,12 +1,52 @@
 rules_version = '2';
 service cloud.firestore {
   match /databases/{database}/documents {
+    function validAmount(docData){
+      return docData.amount is map 
+      && docData.amount.type in ['UNKNOWN', 'FIXED','VARIES','FULL_TUITION']
+      && docData.amount.min is int
+      && docData.amount.max is int
+    }
+
+    function validReqs(docData){
+      return !('requirements' in docData) 
+      || docData.requirements is map
+      && (!('gpa' in docData.requirements) || docData.requirements.gpa is number && docData.requirements.gpa > 0 && docData.requirements.gpa < 4)
+      && (!('ethnicities' in docData.requirements) || docData.requirements.ethnicities is list)
+      && (!('majors' in docData.requirements) || docData.requirements.majors is list)
+      && (!('schools' in docData.requirements) || docData.requirements.schools is list)
+      && (!('grades' in docData.requirements) || docData.requirements.grades is list)
+      && (!('states' in docData.requirements) || docData.requirements.states is list)
+      && (!('genders' in docData.requirements) || docData.requirements.genders is list)
+    }
+
+    function validAuthor(docData){
+      return !('author' in docData) || docData.author is map
```

Comment:
Author should be required

Reply:
I'm assuming the id and email should also be required?